pyiceberg/table/__init__.py

```diff
@@ -593,7 +593,9 @@ def delete(self, delete_filter: Union[str, BooleanExpression], snapshot_properti
                 filtered_df = df.filter(preserve_row_filter)
 
                 # Only rewrite if there are records being deleted
-                if len(df) != len(filtered_df):
+                if len(filtered_df) == 0:
+                    replaced_files.append((original_file.file, []))
+                elif len(df) != len(filtered_df):
```

Comment:
nit: is it more readable if inlined? 
```
if filtered_df and len(df) != len(filtered_df):
```

Reply:
To be honest, I don't see much of a difference.